docs/user_guide/idevid_and_iak.rst

```diff
@@ -0,0 +1,56 @@
+
+How to use 
+---------- 
+
+To allow IDevID and IAK registration, the registrar needs to be, at minimum, loaded with the manufacturer certificates used to provision the IDevID and IAK and configured to accept any TPM identity for registration (the default behaviour). Optionally, the registrar can be configured to **require** an IDevID and IAK in order to register new devices. The agent needs to be configured to turn on IDevID and IAK regeneration, with the correct template selected, and also needs to know where it's IDevID and IAK certificates are stored.
+
+Registrar
+^^^^^^^^^
+
+The certificates for the authority that signed the registering device's IAK and IDevID need to be added to the ``TPM_CERT_STORE``, the default location being ``/var/lib/keylime/tpm_cert_store``.
+
+With the manufacturer certificates added, the Keylime registrar will accept IDevIDs and IAKs received from agents during registration. In this default configuration, the registrar will not prevent for an agent from registering with just an EK instead (and no IDevID and IAK). To disable this behaviour and make the use of IDevID and IAK a requirement for registering devices, the configuration option ``tpm_identity`` needs to be changed:
+
+    tpm_identity = iak_idevid
+
+Agent
+^^^^^
+
+For the agent to use its IDevID and IAK to register, the IAK and IDevID certificates need to be added to the ``keylime_dir`` and named ``iak-cert.crt`` and ``idevid-cert.crt`` respectively. Alternatively, the ``iak_cert`` and ``idevid_cert`` configuration options can be set to absolute paths or filenames relative to the ``keylime_dir``.
+
+Next, the functionality should be enabled by changing ``enable_iak_idevid`` to ``true``. Finally, the template used by the certificates needs to be specified. If the wrong template is chosen, the agent will fail to match the regenerated keys against the imported certificates and registration will fail.
+
+The template should be set using the ``iak_idevid_template`` configuration option. As an alternative, the ``iak_idevid_asymmetric_alg`` and ``iak_idevid_name_alg`` options are available to manually set the algorithms, but this is not recommended, and **these options are ignored** if ``iak_idevid_template`` has been set.
+
+Template definitions can be found in section 7.3.4 of [#tcg]_. If you don't know what template your IAK and IDevID use, the following table can be used to match your algorithms to a template:
+
+==========  ===============  ==========
+ Template    asymmetricAlg    nameAlg
+==========  ===============  ==========
+H-1         RSA 2048         SHA256
+H-2         ECC NIST P256    SHA256
+H-3         ECC NIST P384    SHA384
+H-4         ECC NIST P521    SHA512
+H-5         ECC SM2 P256     SM3_256
+==========  ===============  ==========
```

Comment:
I think users should not have to deal with these. Instead, the agent should read a certificate from a file or later on from an NVRAM location, determine the type (EC vs. RSA) and size of the key and deduce template and nameAlg from it.

Reply:
OK I can work on a patch for that. Would it be worth including the documentation as is for now so people can use the feature as it exists today, then changing when that is added?

Reply:
I propose we merge as is, and a future patch (for the `agent`) is followed by a new PR just to update the documentation. Let's make sure the code capabilities and documentation are on sync.